channels/applicationinsights-channel-js/src/Sender.ts

```diff
@@ -427,20 +439,23 @@ export class Sender extends BaseTelemetryPlugin implements IChannelControlsAI {
             /**
              * error handler
              */
-            _self._onError = (payload: string[], message: string, event?: ErrorEvent) => {
+            _self._onError = (payload: string[], message: string, event?: ErrorEvent, endpointUrl?: string) => {
```

Comment:
How come you need to pass in the `endpointUrl` for every callback? Isn't there someway to access the `Config` globally, so you can just access `endpointUrl` within statsbeat itself?

Reply:
Because for xdr senders, endpointUrl could be different. Eg. for xdr the endpoint url is reprocessed [here](https://github.com/microsoft/ApplicationInsights-JS/blob/44ec248075402dd0827fa8eaeb15e398e9d0be8e/channels/applicationinsights-channel-js/src/Sender.ts#L839). @MSNev please correct me if I understand wrong... or is it ok to ALWYAS get endpointUrl from `config.endpointUrl === undefined ? "https://dc.services.visualstudio.com/v2/track" : config.endpointUrl`  thanks~

Reply:
As the XDR is just ripping off the leading protocol "https", I think we can just always use the config, as this is only for IE9 or lower.